pkg/component/controller/cplb_unix.go

```diff
@@ -274,31 +268,25 @@ func (*Keepalived) getLinkAddresses(link netlink.Link) ([]netlink.Addr, []string
 }
 
 func (k *Keepalived) generateKeepalivedTemplate() error {
-	f, err := os.OpenFile(k.K0sVars.KeepalivedConfigFile, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, fs.FileMode(0500))
-	if err != nil {
-		return fmt.Errorf("failed to open keepalived config file: %w", err)
-	}
-	defer f.Close()
-
-	template, err := template.New("keepalived").Parse(keepalivedConfigTemplate)
-	if err != nil {
-		return fmt.Errorf("failed to parse keepalived template: %w", err)
-	}
-
+	template := template.Must(template.New("keepalived").Parse(keepalivedConfigTemplate))
 	kc := keepalivedConfig{
 		VRRPInstances: k.Config.VRRPInstances,
 	}
-	if err = template.Execute(f, kc); err != nil {
-		return fmt.Errorf("failed to execute keepalived template: %w", err)
-	}
 
-	// TODO: Do we really need to this every single time?
-	if err = os.Chown(k.K0sVars.KeepalivedConfigFile, k.uid, -1); err != nil {
-		return fmt.Errorf("failed to chown keepalived config file: %w", err)
-	}
-	if err = os.Chmod(k.K0sVars.KeepalivedConfigFile, fs.FileMode(0400)); err != nil {
-		return fmt.Errorf("failed to chmod keepalived config file: %w", err)
+	if err := file.WriteAtomically(k.configFilePath, 0400, func(file io.Writer) error {
+		if err := file.(*os.File).Chown(k.uid, -1); err != nil {
+			return err
+		}
+
+		w := bufio.NewWriter(file)
+		if err := template.Execute(w, kc); err != nil {
+			return err
+		}
+		return w.Flush()
```

Comment:
What's the benefit of running it through the bufio.Writer?

Reply:
Buffered writes. I.e. instead of having lots of write calls that write just a couple of bytes directly to disk, buffer them until a reasonable amount of bytes has been accumulated. I think the default is to buffer 4KiB before writing to the wrapped writer (the file on disk in this case). Reduces the number of syscalls and usually speeds up things quite a bit. An alternative would be to buffer everything in memory and write it all at once using a single call to file.Write.